swift/Sources/FlexBuffers/Writer/FlexBuffersWriter.swift

```diff
@@ -0,0 +1,884 @@
+        type: type,
+        bitWidth: bitWidth))
+    return numericCast(sloc)
+  }
+
+  // MARK: Write Vectors
+  @discardableResult
+  @inline(__always)
+  private mutating func create<T>(vector: [T], fixed: Bool) -> Int
+    where T: Scalar
+  {
+    let length: UInt64 = numericCast(vector.count)
+    let vectorType = getScalarType(type: T.self)
+    let byteWidth = MemoryLayout<T>.size
+    let bitWidth = BitWidth.widthB(byteWidth)
+
+    _bb.ensureSpace(size: vector.count &* Int(bitWidth.rawValue))
+
+    assert(widthU(length) <= bitWidth)
+
+    align(width: bitWidth)
+
+    if !fixed {
+      write(value: length, byteWidth: byteWidth)
+    }
+    let vloc = _bb.writerIndex
+
+    for i in stride(from: 0, to: vector.count, by: 1) {
+      write(value: vector[i], byteWidth: byteWidth)
+    }
+
+    stack.append(
+      Value(
+        sloc: .u(numericCast(vloc)),
+        type: toTypedVector(type: vectorType, length: fixed ? length : 0),
+        bitWidth: bitWidth))
+    return vloc
+  }
+
+  @inline(__always)
+  private mutating func createVector(
+    start: Int,
+    count: Int,
+    step: Int,
+    typed: Bool,
+    fixed: Bool,
+    keys: Value? = nil) -> Value
+  {
+    assert(
+      !fixed || typed,
+      "Typed false and fixed true is a combination not supported currently")
+
+    var bitWidth = BitWidth.max(minBitWidth, rhs: widthU(numericCast(count)))
+    var prefixElements = 1
+    if keys != nil {
+      /// If this vector is part of a map, we will pre-fix an offset to the keys
+      /// to this vector.
+      bitWidth = max(bitWidth, keys!.elementWidth(size: writerIndex, index: 0))
+      prefixElements += 2
+    }
+    var vectorType: FlexBufferType = .key
+
+    for i in stride(from: start, to: stack.count, by: step) {
+      let elemWidth = stack[i].elementWidth(
+        size: _bb.writerIndex,
+        index: numericCast(i &- start &+ prefixElements))
+      bitWidth = BitWidth.max(bitWidth, rhs: elemWidth)
+      guard typed else { continue }
+      if i == start {
+        vectorType = stack[i].type
+      } else {
+        assert(
+          vectorType == stack[i].type,
+          """
+          If you get this assert you are writing a typed vector 
+          with elements that are not all the same type
+          """)
+      }
+    }
+    assert(
+      !typed || isTypedVectorElementType(type: vectorType),
+      """
+      If you get this assert, your typed types are not one of:
+      Int / UInt / Float / Key.
+      """)
+
+    let byteWidth = align(width: bitWidth)
+
+    let currentSize: Int = count &* step &* byteWidth
+    let requiredSize: Int = if !typed {
+      // We ensure that we have enough space
+      // for loop two write operations &
+      // 24 bytes for when its not fixed,
+      // and keys isn't null. As an extra safe
+      // guard
+      (currentSize &* 2) &+ twentyFourBytes
+    } else {
+      currentSize
+    }
+
+    _bb.ensureSpace(
+      size: requiredSize)
```

Comment:
I added the following code to ensure that the amount of bytes required would be available to write to, since I was testing with the address sanitizer and there where bugs when a developer would set the size of the buffer to be too small to write to.